projects/core/providers/is-mobile-resolution.provider.ts

```diff
@@ -0,0 +1,29 @@
+import {ElementRef, NgZone, Provider} from '@angular/core';
+import {WINDOW} from '@ng-web-apis/common';
+import {tuiZoneOptimized} from '@taiga-ui/cdk';
+import {TuiMedia} from '@taiga-ui/core/interfaces';
+import {TUI_IS_MOBILE_RES, TUI_MEDIA} from '@taiga-ui/core/tokens';
+import {fromEvent, Observable} from 'rxjs';
+import {distinctUntilChanged, map, startWith} from 'rxjs/operators';
+
+export function isMobileResFactory(
+    windowRef: Window,
+    {mobile}: TuiMedia,
+    {nativeElement}: ElementRef,
+    ngZone: NgZone,
+): Observable<boolean> {
+    const mobile$ = fromEvent(windowRef, 'resize').pipe(
+        startWith(null),
+        map(() => windowRef.innerWidth < mobile),
+        distinctUntilChanged(),
+        tuiZoneOptimized(ngZone),
+    );
+
+    return (nativeElement['$.class._mobile'] = mobile$);
```

Comment:
Isn't that too ninja? Maybe let's use up an extra line but write it clearly :)

Reply:
ok:)